traverse_builtins/ppxlib_traverse_builtins.ml

```diff
@@ -161,6 +162,52 @@ class virtual ['res] lift =
         | x :: l -> self#constr "::" [ f x; self#list f l ]
   end
 
+class virtual ['ctx, 'res] lift_map_with_context =
+  object (self)
+    method virtual other : 'a. 'ctx -> 'a -> 'res
```

Comment:
Maybe it makes sens to have a `lift_with_context` type in `T`, just to increase readability, for methods `other`, `record`, `constr`, `tuple`

Reply:
I don't think we actually get any mileage out of that. I believe those methods show up in four places. In two, they have context arguments; in two they don't. In two (cutting across the other axis) they're virtual; in two they're concrete. So I don't think there's any actual abstraction to be had.